pkg/snapshotmgr/snapshot_manager.go

```diff
@@ -127,6 +128,16 @@ func NewSnapshotManagerFromCluster(params map[string]interface{}, config map[str
 
 func (this *SnapshotManager) CreateSnapshot(peID astrolabe.ProtectedEntityID, tags map[string]string) (astrolabe.ProtectedEntityID, error) {
 	this.Infof("SnapshotManager.CreateSnapshot Called with peID %s, tags %v", peID.String(), tags)
+	return this.createSnapshot(peID, tags, &backupdriverv1.BackupRepository{}, false)
+}
+
+func (this *SnapshotManager) CreateSnapshotWithBackupRepository(peID astrolabe.ProtectedEntityID, tags map[string]string, backupRepository *backupdriverv1.BackupRepository) (astrolabe.ProtectedEntityID, error) {
+	this.Infof("SnapshotManager.CreateSnapshotWithBackupRepository Called with peID %s, tags %v", peID.String(), tags)
+	return this.createSnapshot(peID, tags, backupRepository, true)
+}
+
+func (this *SnapshotManager) createSnapshot(peID astrolabe.ProtectedEntityID, tags map[string]string, backupRepository *backupdriverv1.BackupRepository, useBackupRepository bool) (astrolabe.ProtectedEntityID, error) {
```

Comment:
Is this useBackupRepository necessary?  
Can we just check whether backupRepository is nil?

Reply:
Remove unused booleam useBackupRepository.